libdnf/base/log_event.cpp

```diff
@@ -101,6 +113,8 @@ std::string LogEvent::to_string(
                 throw std::invalid_argument("Incorrect number of elements for ALREADY_INSTALLED");
             }
             return ret.append(utils::sformat(_("Package \"{}\" is already installed."), *additional_data->begin()));
+        case GoalProblem::SOLVER_PROBLEM:
+            return ret.append(solver_problems->to_string());
```

Comment:
`solver_problems` is an `<std::optional>`. I believe the behavior of `solver_problems->` operator is undefined in case it does not contain a value.

Reply:
Correct, but yhe presence is tested on line 124 ` if (!solver_problems) {`. Or did I missed something?